components/google_drive/actions/create-file/create-file.js

```diff
@@ -0,0 +1,244 @@
+    },
+    contentRestrictionsReadOnly: {
+      type: "boolean",
+      label: "Content Restrictions Read Only",
+      description:
+        "Whether the content of the file is read-only. If a file is read-only, a new revision of the file may not be added, comments may not be added or modified, and the title of the file may not be modified.",
+      optional: true,
+    },
+    contentRestrictionsReason: {
+      type: "string",
+      label: "Content Restrictions Reason",
+      description:
+        "Reason for why the content of the file is restricted. This is only mutable on requests that also set readOnly=true.",
+      optional: true,
+    },
+    copyRequiresWriterPermission: {
+      type: "boolean",
+      label: "Copy Requires Writer Permission",
+      description:
+        "Whether the options to copy, print, or download this file, should be disabled for readers and commenters.",
+      optional: true,
+    },
+    description: {
+      type: "string",
+      label: "Description",
+      description: "A short description of the file.",
+      optional: true,
+    },
+    folderColorRgb: {
+      type: "string",
+      label: "Folder Color RGB",
+      description:
+        "The color for a folder as an RGB hex string. If an unsupported color is specified, the closest color in the palette will be used instead.",
+      optional: true,
+    },
+    mimeType: {
+      type: "string",
+      label: "Mime Type",
+      description: `The MIME type of the file. Google Drive will attempt to automatically detect an appropriate value from uploaded content if no value is provided. The value cannot be changed unless a new revision is uploaded. If a file is created with a Google Doc MIME type, the uploaded content will be imported if possible. 
+        Google Workspace and Drive MIME Types: https://developers.google.com/drive/api/v3/mime-types
+      `,
+      optional: true,
+    },
+    name: {
+      type: "string",
+      label: "Name",
+      description: "Name of the file",
+      optional: true,
+    },
+    originalFilename: {
+      type: "string",
+      label: "Original Filename",
+      description:
+        "The original filename of the uploaded content if available, or else the original value of the name field. This is only available for files with binary content in Google Drive.",
+      optional: true,
+    },
+    parents: {
+      type: "string",
+      label: "Parent Folder",
+      description:
+        "The ID of the parent folder which contain the file. If not specified as part of a create request, the file will be placed directly in the user's My Drive folder.",
+      optional: true,
+      async options({ prevContext }) {
+        const { nextPageToken } = prevContext;
+        let results;
+        if (this.drive === "myDrive") {
+          results = await this.googleDrive.listFiles({
+            pageToken: nextPageToken,
+            q: "mimeType = 'application/vnd.google-apps.folder'",
+          });
```

Comment:
Just a question: since we do this same call in other places, would it be a good idea to have a new `listFolders` method in the app module?
```js
module.exports = {
  methods: {
    listFolderOptions(pageToken, opts = {}) {
      return this.listFilesOptions(pageToken, {
        ...opts,
        q: "mimeType = 'application/vnd.google-apps.folder'",
      });
    },
  },
};
```

By the way, we recently pushed a few breaking changes to the app module, so make sure you merge the latest changes in `master` onto this branch.

Reply:
Added listFolders to the app module & merged master